File: php/src/Auth/BearerToken.php
<?php

namespace Auth;

use Psr\Http\Message\ServerRequestInterface;

/**
 * BearerToken implementation for Bearer Token Authentication
 */
class BearerToken extends BaseAuth
{
    private string $token;

    /**
     * Create a new BearerToken instance
     * @param string $token
     */
    public function __construct(string $token)
    {
        $this->type = AuthType::BEARER;
        $this->token = $token;
    }

    /**
     * Authenticate the request using Bearer Token
     * @param ServerRequestInterface $request
     * @return bool
     */
    public function authenticate(ServerRequestInterface $request): bool
    {
        $auth = $request->getHeaderLine('Authorization');
        if (empty($auth)) {
            return false;
        }

        $parts = explode(' ', $auth);
        if (count($parts) !== 2 || $parts[0] !== 'Bearer') {
            return false;
        }

        return $parts[1] === $this->token;
    }

    /**
     * Add Bearer Token to the request
     * @param ServerRequestInterface $request
     * @return ServerRequestInterface
     */
    public function addAuth(ServerRequestInterface $request): ServerRequestInterface
    {
        return $request->withHeader('Authorization', 'Bearer ' . $this->token);
    }
} 

File: go/src/auth/bearer_token.go
package auth

import (
	"net/http"
	"strings"
)

// BearerToken implements the Authenticator interface for Bearer Token Authentication
type BearerToken struct {
	BaseAuth
	Token string
}

// NewBearerToken creates a new BearerToken instance
func NewBearerToken(token string) *BearerToken {
	return &BearerToken{
		BaseAuth: BaseAuth{
			Type: BearerToken,
		},
		Token: token,
	}
}

// Authenticate validates the Bearer Token
func (b *BearerToken) Authenticate(r *http.Request) error {
	auth := r.Header.Get("Authorization")
	if auth == "" {
		return ErrMissingAuth
	}

	parts := strings.Split(auth, " ")
	if len(parts) != 2 || parts[0] != "Bearer" {
		return ErrInvalidAuth
	}

	if parts[1] != b.Token {
		return ErrInvalidCredentials
	}

	return nil
}

// AddAuth adds Bearer Token to the request
func (b *BearerToken) AddAuth(r *http.Request) {
	r.Header.Set("Authorization", "Bearer "+b.Token)
} 